src/telemetry/metrics/transport.rs

```diff
@@ -97,116 +88,107 @@ impl Transports {
         rx: u64,
         tx: u64,
     ) {
-        let key = TransportLabels::new(ctx);
+        let key = Key::new(ctx);
+
+        let metrics = self.metrics.entry(key).or_insert_with(|| Default::default());
```

Comment:
If we're recording a close, and an open didn't exist for a given key, that's probably a bug, right? It seems like this should at least log.

Reply:
this api is going to change